### DISCUSSION TITLE
How to calculate RSRP or Received Power and  Pathloss?

### DISCUSSION THREAD

**Discussion by aibhuyan** [NVlabs/sionna#338] [Q&A] [answered]
I am working with ray tracing scenarios, I need to calculate the RSRP value.
There is some code given:
```
 # Configure antenna array for all transmitters
scene.tx_array = PlanarArray(num_rows=8,
                          num_cols=16,
                          vertical_spacing=0.5,
                          horizontal_spacing=0.5,
                          pattern="dipole",
                          polarization="V")

# Configure antenna array for all receivers
scene.rx_array = PlanarArray(num_rows=1,
                          num_cols=1,
                          vertical_spacing=0.5,
                          horizontal_spacing=0.5,
                          pattern="dipole",
                          polarization="V")

# Create transmitter
tx = Transmitter(name="tx",
              position=[57,13,15],
              orientation=[0,0,0])
scene.add(tx)

# Create a receiver
rx = Receiver(name="rx",
           position=[13,4,1.5],
           orientation=[0,0,0])
scene.add(rx)
```

**Now how can I calculate RSRP?**
We can get paths by using:
`paths = scene.compute_paths()`

**How do you get the pathloss now?**

**Another question is how can I set the camera for rendering to view the overall scenario?**
In the given screenshot the overview is not looking nicely.
<img width="586" alt="1j" src="https://github.com/NVlabs/sionna/assets/149014413/fc7a3019-78b1-4421-b735-c7ef7760767f">

**Comment by faycalaa** [comment]
Hi,

You can calculate the path loss incurred by the channel from the [path coefficients](https://nvlabs.github.io/sionna/api/rt.html#sionna.rt.Paths.a) by summing their squared amplitudes over the path dimension. In scenarios involving multiple antennas, it might be necessary to consider the precoding and/or combining vectors.

Note that if you are only interested in the path loss, you could also consider computing a [coverage map](https://nvlabs.github.io/sionna/api/rt.html#coverage-map).

Regarding camera settings, you need to configure its [position and orientation](https://nvlabs.github.io/sionna/api/rt.html#cameras). You could also open a [preview](https://nvlabs.github.io/sionna/api/rt.html#sionna.rt.Scene.preview) and render from the viewpoint of this preview as shown below:

```
scene.preview() # Open preview showing paths
scene.render(camera="preview") # Render from preview camera
```

Hope this helps.

**Reply by aibhuyan** [reply, marked as answer]
Thanks, @faycalaa  for your answer.

**Reply by luannt0801** [reply]
Hi @faycalaa,

If we calculate the received power by summing all propagation paths, this corresponds to RSS, not RSRP.
As far as I understand, RSRP is derived only from reference signals, not from the total received signal power. Specifically, RSRP is computed by averaging the received power over the resource blocks (or resource elements) that carry reference signals at the UE.

Is there any method to calculate or simulate the number of resource blocks (or reference-signal-bearing resources) received at the UE side?

Thanks.
